packages/nextra-theme-docs/src/index.tsx

```diff
@@ -22,6 +22,20 @@ import './polyfill'
 import Breadcrumb from './breadcrumb'
 import renderComponent from './utils/render-component'
 
+let resizeObserver: ResizeObserver
+if (typeof window !== 'undefined') {
+  resizeObserver =
+    resizeObserver! ||
+    new ResizeObserver(entries => {
+      if (window.location.hash) {
+        entries[0]
+        .target
+        .ownerDocument
+        .querySelector(window.location.hash)?.scrollIntoView()
```

Comment:
I think you could use `scroll-into-view-if-needed` here since it's already installed in docs theme